### ISSUE TITLE
Function parameters referenced in closure not properly handled by no-unused-vars

### ISSUE THREAD

**Issue by burnnat** [eslint/eslint#584]
Consider the following javascript:

``` javascript
var doStuff = function(fn) {
    fn();
};

var callStuff = function(first, second) {
    doStuff(function() {
        console.log(second);
    });
};

callStuff('foo', 'bar');
```

Running ESLint on this sample code yields an error from the `no-unused-vars` rule:

```
sample.js: line 5, col 25, Error - first is defined but never used (no-unused-vars)

1 problem
```

Normally, the `no-unused-vars` rule will ignore any function parameters that precede a "used" parameter. But this seems to break when the parameter is used inside a closure, as in this example - even though the `second` parameter is properly detected as being used, the `first` parameter is not ignored and causes a spurious error.
